### PR TITLE
Introduces Github Actions CI

### PR DESCRIPTION
**Pull request by kelsos**
Testing out #809

It should include all the functionality that is currently in the Travis configuration
- Lint backend
- Lint/test/build frontend
- Test backend on Linux/OSX
- Test integration on Linux only (same issues here as with travis on OSX)
- Packaging on tag (tested that the build happens)

A successful run here: https://github.com/kelsos/rotki/actions/runs/72906412
The time is because some tests kept failing and I was restarting.

We can enable it as an experiment, but not require it and see if it makes sense to move to it or not.

### CODE REVIEW

File: rotkehlchen/tests/external_apis/test_etherscan.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def _handle_killed_greenlets(greenlet: gevent.Greenlet) -> None:
 
 
 @pytest.mark.skipif(
-    'TRAVIS' in os.environ,
+    'CI' in os.environ,
```
Comment (LefterisJP) [inline rotkehlchen/tests/external_apis/test_etherscan.py:48]: Guess you checked and it does but this is both in Circle and Travis, right?

File: .github/workflows/rotki_ci.yml [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            ${{ runner.os }}-pip-
+      - name: Install dependencies
+        run: |
+          pip install --upgrade "pip<19.0.0" wheel
+          pip install pytest-travis-fold codecov pytest-cov
+          pip install -r requirements_dev.txt
+          pip install -e .
+      - name: Run Test
+        run: |
+          COVERAGE_ARGS='--cov=./ --travis-fold=always'
```
Comment (LefterisJP) [inline .github/workflows/rotki_ci.yml:122]: In Travis we were invoking some scripts. Can the same not happen here?

File: rotkehlchen/tests/external_apis/test_etherscan.py (hunk 1 above)
Comment (kelsos) [reply rotkehlchen/tests/external_apis/test_etherscan.py:48]: Yes CI is by default true on travis

**Comment by kelsos** [comment]
In Windows I am still trying to figure the script to compile steps for sqlcipher and pysqlchiper, I had no luck so far.

**Comment by LefterisJP** [comment]
> In Windows I am still trying to figure the script to compile steps for sqlcipher and pysqlchiper, I had no luck so far.

Hmm how can I help there? Can naturally be done in a different PR.

File: .github/workflows/rotki_ci.yml (hunk 2 above)
Comment (kelsos) [reply .github/workflows/rotki_ci.yml:122]: I will have to modify the script to make it compatible with GitHub Actions due to the following
https://github.com/rotki/rotki/blob/9f76312ad7552e2d35b6342e2ba3a3f13880fc98/.travis/run_tests.sh#L7

**Comment by kelsos** [comment]
Yes, it can be done in a different PR. Mostly I need to figure out how to set the environment for `nmake` to run inside a batch script, which doesn't seem to be as straightforward as I would like.

File: .github/workflows/rotki_ci.yml (hunk 2 above)
Comment (LefterisJP) [reply .github/workflows/rotki_ci.yml:122]: I see. No need to do so. Just wanted to know if the syntax allowed it

**Pull request merged**